Clang rejects the template keyword before a name with no template argument
list (-Wmissing-template-arg-list-after-template-kw). call_helper is an
ordinary static member function, so the template disambiguator is invalid.

diff --git a/include/boost/msm/front/row2.hpp b/include/boost/msm/front/row2.hpp
--- a/include/boost/msm/front/row2.hpp
+++ b/include/boost/msm/front/row2.hpp
@@ -51,7 +51,7 @@
                                 AllStates& all_states)
         {
             // in this front-end, we don't need to know source and target states
-            ::boost::msm::front::detail::row2_action_helper<CalledForAction,Event,action>::template call_helper
+            ::boost::msm::front::detail::row2_action_helper<CalledForAction,Event,action>::call_helper
                 (fsm,evt,src,tgt,all_states,
                 ::boost::mpl::bool_< ::boost::is_base_of<CalledForAction,FSM>::type::value>());
             return ::boost::msm::back::HANDLED_TRUE;
